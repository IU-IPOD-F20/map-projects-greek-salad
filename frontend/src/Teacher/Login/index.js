import React from "react";
import { Form, Input, Button, Checkbox, Typography } from "antd";
import { useHistory } from "react-router-dom";

import "./style.css";

const layout = {
  labelCol: {
    span: 8,
  },
  wrapperCol: {
    span: 16,
  },
};
const tailLayout = {
  wrapperCol: {
    offset: 8,
    span: 16,
  },
};

const { Title } = Typography;

const Login = () => {
  const history = useHistory();

  const [isLogin, onLogin] = React.useState(true);

  const onFinish = (values) => {
    console.log("Success:", values);
    history.push("/teacher");
  };

  const onFinishFailed = (errorInfo) => {
    console.log("Failed:", errorInfo);
  };

  return (
    <div className="login">
      {isLogin ? (
        <Title style={{ textAlign: "center" }}>Login</Title>
      ) : (
        <Title style={{ textAlign: "center" }}>Signup</Title>
      )}
      <Form
        {...layout}
        name="basic"
        initialValues={{
          remember: true,
        }}
        onFinish={onFinish}
        onFinishFailed={onFinishFailed}
        style={{
          display: "flex",
          flexDirection: "column",
          width: "40vw",
          margin: "0 auto",
        }}
      >
        <Form.Item
          label="Username"
          name="username"
          rules={[
            {
              required: true,
              message: "Please input your username!",
            },
          ]}
        >
          <Input />
        </Form.Item>

        <Form.Item
          label="Password"
          name="password"
          rules={[
            {
              required: true,
              message: "Please input your password!",
            },
          ]}
        >
          <Input.Password />
        </Form.Item>

        <Form.Item {...tailLayout} name="remember" valuePropName="checked">
          <Checkbox>Remember me</Checkbox>
        </Form.Item>

        <Form.Item {...tailLayout}>
          <Button type="primary" htmlType="submit">
            Submit
          </Button>
        </Form.Item>
      </Form>
    </div>
  );
};

export default Login;
