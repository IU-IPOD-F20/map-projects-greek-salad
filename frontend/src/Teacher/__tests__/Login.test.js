import React from "react";
import { MemoryRouter } from "react-router";
import Login from "../Login";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

test("login initial render app", () => {
  const wrapper = shallow(
    <MemoryRouter initialEntries={["/login"]}>
      <Login />
    </MemoryRouter>
  );
  expect(wrapper).toMatchSnapshot();
});
