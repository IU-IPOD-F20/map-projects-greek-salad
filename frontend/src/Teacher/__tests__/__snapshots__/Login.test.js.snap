// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`student initial render app 1`] = `ShallowWrapper {}`;
